### PR TITLE
Onnx support and doc

### PR DESCRIPTION
**Pull request by qibaoyuan**
## Motivation and Context

I have use org.bytedeco.javacpp-presets and deeplearning4j in my JAVA application to do Deep learning task in production env, and it can use onnx as a pre-trained model. So, I make this PR to copy the model as a .onnx file after it was converted from pytorch-type model.I make a "export_onnx_path" para, in this PR I split "--output-model-path" to "--output-caffe2-path" and "--output-onnx-path" in "export" command.

## How Has This Been Tested

my test code under JAVA applciation in javacpp-presets is as belows:

```java
onnx.OpSchemaVector allSchemas = onnx.OpSchemaRegistry.get_all_schemas();
System.out.println(allSchemas.size());

byte[] bytes = 
 Files.readAllBytes(Paths.get("/Users/qibaoyuan/Documents/source/project/pycharm/pytext/minlpdemo/ner/model.pt.prd.2.onnx"));

onnx.ModelProto model = new onnx.ModelProto();
ParseProtoFromBytes(model, new BytePointer(bytes), bytes.length);

check_model(model);

InferShapes(model);

onnx.StringVector passes = new onnx.StringVector("eliminate_nop_transpose", "eliminate_nop_pad", "fuse_consecutive_transposes", "fuse_transpose_into_gemm");
Optimize(model, passes);

check_model(model);

ConvertVersion(model, 8);

System.out.println(model.graph().input_size());
```
or in deeplearning4j:
```java
    public static OnnxProto3.GraphProto loadModel(final String file) {
        try (InputStream is = new ClassPathResource(file).getInputStream()) {
            OnnxProto3.GraphProto graphProto = OnnxProto3.GraphProto.parseFrom(is);
            log.debug("load model from onnx:[{}]", graphProto);
            return graphProto;
        } catch (Exception ex) {
            log.error("error:[{}]", ExceptionUtils.getStackTrace(ex));
        }
        return null;

    }
```

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: pytext/main.py [hunk 1]
```diff
@@ -271,15 +271,21 @@ def train(context):
...
 @click.pass_context
-def export(context, model, output_path):
+def export(context, model, output_onnx_path, output_caffe2_path):
     """Convert a pytext model snapshot to a caffe2 model."""
     config = context.obj.load_config()
-    model = model or config.save_snapshot_path
-    output_path = output_path or config.export_caffe2_path
-    print(f"Exporting {model} to {output_path}")
-    export_saved_model_to_caffe2(model, output_path)
+    model = model or config.save_snapshot_path0
```
Comment (seayoung1112) [inline pytext/main.py]: path0, typo?

File: pytext/task/task.py [hunk 2]
```diff
@@ -168,12 +168,13 @@ def test(self, test_path):
         test_iter = self.data_handler.get_test_iter()
         return self.trainer.test(test_iter, self.model, self.metric_reporter)
 
-    def export(self, model, export_path, summary_writer=None):
+    def export(self, model, export_onnx_path, export_caffe2_path, summary_writer=None):
```
Comment (seayoung1112) [inline pytext/task/task.py]: same here

File: pytext/utils/onnx_utils.py [hunk 3]
```diff
@@ -32,12 +39,14 @@ def pytorch_to_caffe2(
     torch.onnx.export(
         model,
         export_input,
-        export_path,
+        export_onnx_path or export_caffe2_path,
```
Comment (seayoung1112) [inline pytext/utils/onnx_utils.py]: now that we already have export_onnx_path, just always export to it to make it clear

File: pytext/main.py [hunk 4]
```diff
@@ -271,15 +271,21 @@ def train(context):
 
 @main.command()
 @click.option("--model", help="the pytext snapshot model file to load")
-@click.option("--output-path", help="where to save the exported model")
+@click.option("--output-onnx-path", help="where to save the exported onnx model")
+@click.option("--output-caffe2-path", help="where to save the exported caffe2 model")
```
Comment (seayoung1112) [inline pytext/main.py]: please keep the existing param name, as users are already using it

**Comment by qibaoyuan** [comment]
@seayoung1112 I make a new PR in https://github.com/facebookresearch/pytext/pull/187 , I have made some fixes according to what you have mentioned. However, in onnx_utils.py the parameter 'export_onnx_path' is considered as an optional, so the onnx file will not be created once it's not assigned in an explicit way.

**Pull request closed without merging**